packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -261,14 +254,8 @@ contract ERC20Vault is BaseVault {
 
         if (ctoken.addr == address(0)) revert VAULT_INVALID_TOKEN();
 
-        if (amount > 0) {
-            if (bridgedToCanonical[ctoken.addr].addr != address(0)) {
-                IBridgedERC20(ctoken.addr).mint(message.owner, amount);
-            } else {
-                ERC20(ctoken.addr).safeTransfer(message.owner, amount);
-            }
-        }
-
+        // Transfer the ETH and tokens back to the owner
+        _transferTokens(ctoken, message.owner, amount);
```

Comment:
same, maybe catch the token address return value and emit in the `TokenReleased` ?